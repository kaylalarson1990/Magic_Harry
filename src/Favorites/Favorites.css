.favorites {
    font-size: 48px;
    font-weight: 400;
    color: #BE0125;
}