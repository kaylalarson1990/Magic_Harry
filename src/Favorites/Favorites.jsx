import React from "react";
import Card from "../Card/Card.jsx";
import "./Favorites.css";
import PropTypes from "prop-types";

const Favorites = ({ favorites, favoriteCard }) => {
  const showFavorites = favorites.map(fav => {
    return <Card info={fav} key={fav.created} favoriteCard={favoriteCard} />;
  });

  return (
    <article>
      <h2 className="title">Favorites</h2>
      <section className="fav-container">
        {!favorites.length ? (
          <h2 className="favorites">
            <i /> No Favorites Available
          </h2>
        ) : (
          showFavorites
        )}
      </section>
    </article>
  );
};

Favorites.propTypes = {
  favorites: PropTypes.array,
  favoriteCard: PropTypes.func
};

export default Favorites;
