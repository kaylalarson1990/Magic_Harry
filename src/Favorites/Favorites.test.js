import React from "react";
import { shallow } from "enzyme";
import Favorites from "./Favorites.jsx";

describe("Favorites", () => {
  let wrapper, mockData;

  beforeEach(() => {
    mockData = [
      {
        favorites: "Kayla"
      }
    ];
    wrapper = shallow(
      <Favorites favorites={mockData} favoriteCard={jest.fn()} />
    );
  });
  xit("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
