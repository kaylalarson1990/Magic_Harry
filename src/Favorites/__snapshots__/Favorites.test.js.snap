// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites should match snapshot 1`] = `
<article>
  <h2
    className="title"
  >
    Favorites
  </h2>
  <section
    className="fav-container"
  >
    <Card
      favoriteCard={[MockFunction]}
      info={
        Object {
          "favorites": "Kayla",
        }
      }
    />
  </section>
</article>
`;
